pkg/metadata/annotations/meta.go

```diff
@@ -0,0 +1,22 @@
+package annotations
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+func Remove(annotation Annotation) OptionFunc {
+	return func(obj metav1.Object) {
+		existingAnnotations := obj.GetAnnotations()
+		if existingAnnotations == nil {
+			existingAnnotations = make(map[string]string)
+		}
+
+		delete(existingAnnotations, annotation.Key())
+
+		obj.SetAnnotations(existingAnnotations)
+	}
+}
+
+type OptionFunc func(obj metav1.Object)
+
+func (f OptionFunc) ApplyToMeta(obj metav1.Object) {
+	f(obj)
+}
```

Comment:
It's better if this is in `metadata` package as it's not specific to annotation. I deliberately had `metadata.OptionFunc` in the original func definition :)

Reply:
I have should have trusted the comment : ) [5e94df6](https://github.com/maistra/odh-platform/pull/48/commits/5e94df6de41e541721ec236a007dfa6898357b80)